### PR TITLE
Remaining memory management fixes of RT #3198 (1.0.2 version)

### PR DESCRIPTION
**Pull request by mattcaswell**
This is a backport to 1.0.2 of the changes in #1161 that are relevant to that branch.

### CODE REVIEW

File: crypto/engine/eng_cryptodev.c [hunk 1]
```diff
@@ -1065,7 +1065,7 @@ static void zapparams(struct crypt_kop *kop)
 
     for (i = 0; i < kop->crk_iparams + kop->crk_oparams; i++) {
         if (kop->crk_param[i].crp_p)
-            free(kop->crk_param[i].crp_p);
+            OPENSSL_free(kop->crk_param[i].crp_p);
```
Comment (richsalz) [inline crypto/engine/eng_cryptodev.c]: remove the if on the previous line.

File: crypto/engine/eng_cryptodev.c [hunk 2]
```diff
@@ -1078,16 +1078,24 @@ cryptodev_asym(struct crypt_kop *kop, int rlen, BIGNUM *r, int slen,
...
+        kop->crk_param[kop->crk_iparams].crp_p = OPENSSL_zalloc(rlen);
+        if (kop->crk_param[kop->crk_iparams].crp_p == NULL)
+            return ret;
         kop->crk_param[kop->crk_iparams].crp_nbits = rlen * 8;
         kop->crk_oparams++;
     }
     if (s) {
         kop->crk_param[kop->crk_iparams + 1].crp_p =
-            calloc(slen, sizeof(char));
+            OPENSSL_zalloc(slen);
```
Comment (richsalz) [inline crypto/engine/eng_cryptodev.c]: zalloc doesn't exist in 1.0.2

**Comment by mattcaswell** [comment]
Comments address and updated commits pushed. Tested to build and tests run on Linux and Freebsd.

**Comment by mattcaswell** [comment]
Pushed. Thanks

**Pull request closed without merging**